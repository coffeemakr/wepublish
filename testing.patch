diff --git a/libs/api/src/lib/db/article.ts b/libs/api/src/lib/db/article.ts
index 7359af6a..ea27be09 100644
--- a/libs/api/src/lib/db/article.ts
+++ b/libs/api/src/lib/db/article.ts
@@ -14,11 +14,11 @@ export interface ArticleData {
   readonly lead?: string | null
   readonly seoTitle?: string | null
   readonly slug?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
   readonly canonicalUrl?: string | null
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
   readonly authors: ArticleRevisionAuthor[]
@@ -107,7 +107,7 @@ export type ArticleRevisionWithRelations = PrismaArticleRevision & {
 
 export type ArticleWithRevisions = PrismaArticle & {
   draft: ArticleRevisionWithRelations | null
-  pending: ArticleRevisionWithRelations | null
+  pending?: ArticleRevisionWithRelations | null
   published: ArticleRevisionWithRelations | null
 }
 
diff --git a/libs/api/src/lib/db/page.ts b/libs/api/src/lib/db/page.ts
index fb73c289..6f0ed1b5 100644
--- a/libs/api/src/lib/db/page.ts
+++ b/libs/api/src/lib/db/page.ts
@@ -14,9 +14,9 @@ export interface PageData {
 
   readonly title: string
   readonly description?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
 
@@ -85,7 +85,7 @@ export type PageRevisionWithProperties = PrismaPageRevision & {
 
 export type PageWithRevisions = PrismaPage & {
   draft: PageRevisionWithProperties | null
-  pending: PageRevisionWithProperties | null
+  pending?: PageRevisionWithProperties | null
   published: PageRevisionWithProperties | null
 }
 
diff --git a/libs/api/src/lib/graphql/article.ts b/libs/api/src/lib/graphql/article.ts
index 8d80fb27..77357e01 100644
--- a/libs/api/src/lib/graphql/article.ts
+++ b/libs/api/src/lib/graphql/article.ts
@@ -275,7 +275,7 @@ export const GraphQLPublicArticle: GraphQLObjectType<PublicArticle, Context> =
         type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
         resolve: ({properties}) => {
           return properties
-            .filter(property => property.public)
+            ?.filter(property => property.public)
             .map(({key, value}) => ({key, value}))
         }
       },
diff --git a/libs/api/src/lib/graphql/article/article.private-mutation.ts b/libs/api/src/lib/graphql/article/article.private-mutation.ts
index 43a6a3ee..049c7e54 100644
--- a/libs/api/src/lib/graphql/article/article.private-mutation.ts
+++ b/libs/api/src/lib/graphql/article/article.private-mutation.ts
@@ -178,7 +178,8 @@ export const duplicateArticle = async (
       createMany: {
         data: socialMediaAuthors.map(({authorId}) => ({authorId}))
       }
-    }
+    },
+    blocks: []
   }
 
   return articleClient.create({
@@ -303,7 +304,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -333,7 +335,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -491,7 +494,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -521,7 +525,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -579,7 +584,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -609,7 +615,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
diff --git a/libs/api/src/lib/graphql/page.ts b/libs/api/src/lib/graphql/page.ts
index 56d380f0..6b63518e 100644
--- a/libs/api/src/lib/graphql/page.ts
+++ b/libs/api/src/lib/graphql/page.ts
@@ -210,7 +210,7 @@ export const GraphQLPublicPage = new GraphQLObjectType<PublicPage, Context>({
     properties: {
       type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
       resolve: ({properties}) => {
-        return properties.filter(property => property.public).map(({key, value}) => ({key, value}))
+        return properties?.filter(property => property.public).map(({key, value}) => ({key, value}))
       }
     },
 
diff --git a/libs/api/src/lib/graphql/page/page.private-mutation.ts b/libs/api/src/lib/graphql/page/page.private-mutation.ts
index 8d21a995..456df84e 100644
--- a/libs/api/src/lib/graphql/page/page.private-mutation.ts
+++ b/libs/api/src/lib/graphql/page/page.private-mutation.ts
@@ -140,7 +140,8 @@ export const duplicatePage = async (
       createMany: {
         data: duplicatedProperties
       }
-    }
+    },
+    blocks: []
   }
 
   return pageClient.create({
@@ -236,7 +237,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -250,7 +252,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -387,7 +390,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -401,7 +405,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -443,7 +448,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -454,7 +460,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
