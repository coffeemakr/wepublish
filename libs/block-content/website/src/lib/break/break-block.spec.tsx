import {runStorybookTests} from '@wepublish/testing'
import * as stories from './break-block.stories'

describe('Break Block', () => {
  runStorybookTests(stories)
})
