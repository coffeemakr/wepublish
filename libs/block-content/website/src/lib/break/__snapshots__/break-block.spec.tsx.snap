// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Break Block should render Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  gap: 16px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:900px) {
  .emotion-0 {
    grid-template-columns: 1fr 1fr;
    gap: 40px;
  }
}

.emotion-2 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 16px;
}

.emotion-4 {
  max-width: 100%;
  width: 100%;
  height: auto;
  aspect-ratio: auto 0.6666666666666666;
}

@media (max-width:899.95px) {
  .emotion-4 {
    aspect-ratio: 1;
  }
}

.emotion-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
}

.emotion-11 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  margin-bottom: 0.35em;
}

.emotion-13 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <img
        alt="DSC07717"
        class="emotion-4 emotion-5"
        loading="lazy"
        srcset="https://unsplash.it/800/400 800w,
https://unsplash.it/500/300 500w,
https://unsplash.it/300/200 300w,
https://unsplash.it/200/100 200w"
        title=""
      />
    </div>
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="MuiTypography-root MuiTypography-h4 emotion-8"
        role="heading"
      >
        Break block test
      </div>
      <div
        class="emotion-9 emotion-10"
      >
        <div
          data-gramm="false"
          data-slate-editor="true"
          data-slate-node="value"
          style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-11"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-11"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </p>
          <h4
            class="MuiTypography-root MuiTypography-h6 emotion-13"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-string="true"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit
              </span>
            </span>
          </h4>
          <h4
            class="MuiTypography-root MuiTypography-h6 emotion-13"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-length="0"
                data-slate-zero-width="n"
              >
                ﻿
                <br />
              </span>
            </span>
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom emotion-11"
            data-slate-node="element"
          >
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <strong
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                  </span>
                </strong>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <em
                data-slate-leaf="true"
              >
                <span
                  data-slate-string="true"
                >
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit
                </span>
              </em>
            </span>
            <span
              data-slate-node="text"
            >
              <u
                data-slate-leaf="true"
              >
                <em
                  data-slate-leaf="true"
                >
                  <span
                    data-slate-string="true"
                  >
                    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                  </span>
                </em>
              </u>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Break Block should render ImageRight 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,11 +1,11 @@
  <DocumentFragment>
    <div
      class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
    >
      <div
-       class="css-ywr472-BreakBlockSegment e11xbw3q0"
+       class="css-17wr6dm-BreakBlockSegment e11xbw3q0"
      >
        <img
          alt="DSC07717"
          class="css-16m3onx-ImageWrapper-BreakBlock e1shv2yg0"
          loading="lazy""
`;

exports[`Break Block should render WithButton 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -146,8 +146,18 @@
                </u>
              </span>
            </p>
          </div>
        </div>
+       <a
+         class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-xzjkiw-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
+         href="Link URL"
+         tabindex="0"
+       >
+         Link Text
+         <span
+           class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
+         />
+       </a>
      </div>
    </div>
  </DocumentFragment>"
`;

exports[`Break Block should render WithClassName 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <DocumentFragment>
    <div
-     class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
+     class="extra-classname css-1k98guj-BreakBlockWrapper e11xbw3q1"
    >
      <div
        class="css-ywr472-BreakBlockSegment e11xbw3q0"
      >
        <img"
`;

exports[`Break Block should render WithEmotion 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Break Block should render WithoutImage 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -3,30 +3,20 @@
      class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
    >
      <div
        class="css-ywr472-BreakBlockSegment e11xbw3q0"
      >
-       <img
-         alt="DSC07717"
-         class="css-16m3onx-ImageWrapper-BreakBlock e1shv2yg0"
-         loading="lazy"
-         srcset="https://unsplash.it/800/400 800w,
- https://unsplash.it/500/300 500w,
- https://unsplash.it/300/200 300w,
- https://unsplash.it/200/100 200w"
-         title=""
-       />
-     </div>
-     <div
-       class="css-ywr472-BreakBlockSegment e11xbw3q0"
-     >
        <div
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          role="heading"
        >
          Break block test
+       </div>
      </div>
+     <div
+       class="css-ywr472-BreakBlockSegment e11xbw3q0"
+     >
        <div
          class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
        >
          <div
            data-gramm="false""
`;

exports[`Break Block should render WithoutImageRight 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,32 +1,22 @@
  <DocumentFragment>
    <div
      class="css-1k98guj-BreakBlockWrapper e11xbw3q1"
    >
      <div
-       class="css-ywr472-BreakBlockSegment e11xbw3q0"
-     >
-       <img
-         alt="DSC07717"
-         class="css-16m3onx-ImageWrapper-BreakBlock e1shv2yg0"
-         loading="lazy"
-         srcset="https://unsplash.it/800/400 800w,
- https://unsplash.it/500/300 500w,
- https://unsplash.it/300/200 300w,
- https://unsplash.it/200/100 200w"
-         title=""
-       />
-     </div>
-     <div
-       class="css-ywr472-BreakBlockSegment e11xbw3q0"
+       class="css-17wr6dm-BreakBlockSegment e11xbw3q0"
      >
        <div
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          role="heading"
        >
          Break block test
        </div>
+     </div>
+     <div
+       class="css-ywr472-BreakBlockSegment e11xbw3q0"
+     >
        <div
          class="css-1v4yto8-RichTextBlockWrapper eu4zjs90"
        >
          <div
            data-gramm="false""
`;
